core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/XmlProtocolUnmarshaller.java

```diff
@@ -79,19 +90,19 @@ public <TypeT extends SdkPojo> TypeT unmarshall(SdkPojo sdkPojo,
     }
 
     SdkPojo unmarshall(XmlUnmarshallerContext context, SdkPojo sdkPojo, XmlElement root) {
```

Comment:
The nesting in this method is pretty wild. Can we flatten it down to a single nesting within the `for (SdkField<?> field : sdkPojo.sdkFields()) {` loop, using `continue;` statements? 

For example, remove the nesting for payload handling with:
```java
if (field.location() != MarshallLocation.PAYLOAD) {
    Object unmarshalled = unmarshaller.unmarshall(context, null, (SdkField<?>) field);
    field.set(sdkPojo, unmarshalled);
    continue;
}

// Handle field.location() == MarshallLocation.PAYLAD
```

and remove the nesting for attribute handling with:

```java
if (isAttribute(field)) {
    root.getOptionalAttributeByName(field.unmarshallLocationName())
        .ifPresent(e -> field.set(sdkPojo, e));
    continue;
}

// Handle isAttribute(field) == false
```

Reply:
sounds good, will flatten down